### ISSUE TITLE
wx.dataview.DataViewTreeCtrl crash when using data association to elements

### ISSUE THREAD

**Issue by xader** [wxWidgets/Phoenix#856]
**Operating system**: Windows 10
**wxPython version**: 4.0.1
**Stock or custom build**: stock
**Python version**: 3.6.3
**Stock or custom build**: Stock

**Description of the problem**:
crash during exit or by explictly calling DeleteAllItems

```python
import wx
import wx.dataview
if __name__ == '__main__':
    app = wx.App()
    frame = wx.Frame(None,-1)
    panel = wx.Panel(frame)
    sizer = wx.BoxSizer(wx.VERTICAL)
    panel.SetSizer(sizer)
    treectrl = wx.dataview.DataViewTreeCtrl(panel)
    sizer.Add(treectrl, -1, wx.EXPAND)
    data =  wx.ClientDataContainer()
    data.SetClientData('test data')
    treectrl.AppendContainer(wx.dataview.NullDataViewItem,'test', data=data)
    frame.Show()
    
    app.MainLoop()
```

**Comment by RobinD42** [comment]
It looks there are some misunderstandings here. 

1. `wx.ClientDataContainer` is not meant to be used this way. It's used as the common base class for other classes that can have client data. IOW, it's  a convenient way to add Get/SetClientData methods to a bunch of other classes. I think it's mostly used in classes in the `wxGrid` family.

2. `wx.dataview.DataViewTreeCtrl.AppendContainer` does not mean "append an item with client data" it means "append an item that has (or *contains*) child items."

3. wxPython has special support for the C++ `wxClientData` class which allows any Python object to be passed in and it will take care of things like the refcount and such. So you can use anything there, like `data="test data"` or "data=["test", "data"]` or `data=someObject` or whatever.  However it is still crashing when data is assigned to an item, so I'll dig into that and get it fixed.

**Comment by RobinD42** [comment]
Fixed in #858
